radis/test/lbl/test_factory.py

```diff
@@ -644,6 +644,140 @@ def test_all_spectrum_using_wstep_auto(verbose=True, plot=False, *args, **kwargs
     assert sf._wstep == "auto"
 
 
+# Added in https://github.com/radis/radis/pull/580
+# Compares spectrum calculated using vaex and pandas are same.CONVOLUTED_QUANTITIES and dataframe df is compared to ensure the spectrum is same.
+def test_vaex_and_pandas_spectrum():
+    from radis import calc_spectrum
+    from radis.spectrum.utils import CONVOLUTED_QUANTITIES
+
+    # Comparison is made for all the databanks to ensure that calculations are consistent across databanks and maximum part of code is tested.
+    databanks = ["hitran", "hitemp", "exomol", "geisa"]
+
+    for databank in databanks:
+        # computing spectrum in vaex dataframe format
+        s, factory_s = calc_spectrum(
+            1900,
+            2300,  # cm-1
+            molecule="CO",
+            isotope="1,2,3",
+            pressure=1.01325,  # bar
+            Tgas=700,  # K
+            # mole_fraction={'CO': 0.5, 'CO2': 0.3},
+            mole_fraction=0.1,
+            path_length=1,  # cm
+            databank=databank,  # or 'hitemp', 'geisa', 'exomol'
+            # use_cached=False,
+            engine="vaex",
+            return_factory=True,
+        )
+        s.apply_slit(0.5, "nm")  # simulate an experimental slit
+
+        # computing spectrum in pandas dataframe format
+        s1, factory_s1 = calc_spectrum(
+            1900,
+            2300,  # cm-1
+            molecule="CO",
+            isotope="1,2,3",
+            pressure=1.01325,  # bar
+            Tgas=700,  # K
+            mole_fraction=0.1,
+            path_length=1,  # cm
+            databank=databank,  # or 'hitemp', 'geisa', 'exomol'
+            # use_cached=False,
+            engine="pandas",
+            return_factory=True,
+        )
+        s1.apply_slit(0.5, "nm")  # simulate an experimental slit
+
+        for spec_quantity in CONVOLUTED_QUANTITIES:
```

Comment:
s1.compare_with(s)  should work here ; doesn't it ?

Reply:
s1.compare_with works when data frames are of the same type, here data frames are of different types.